envoy-auth-layer/README.md

```diff
@@ -0,0 +1,169 @@
+# Readme
+
+This is a POC auth-layer for Hedera-The-Graph implementation that will allow a node operator to publish a secured `admin port` of the-graph deployment for hedera.
+
+Uses EnvoyProxy as a reverse proxy that handles the token verification (auth layer)
+
+```mermaid
+---
+title: Overall Architecture
+---
+flowchart LR
+    A[Graph Cli] --->|1. Create / Deploy Subgraph| B(Auth-Layer)
+    B ---> | 4. Relay Request on Admin Port| F(TheGraph Service)
+    B --> | 3. 401 Unauthorized | A
+    F ---> | 5. response| B --> | 6. response| A
+    B <--> | 2. Check Token| db[(Database)]
+    
+    linkStyle 2 stroke:#ff0000,stroke-width:2px;
+    linkStyle 3 stroke:#00ff00,stroke-width:2px;
+    linkStyle 4 stroke:#00ff00,stroke-width:2px;
+
+```
+
+## Overview
+
+This is an implementation of EnvoyProxy filters for authentication and authorization. It is a custom config with http filters using Lua scripts for performing the following actions: 
+
+1. JSON Validation
+2. Token Extraction
+3. Payload Params Extraction
+3. Token Validation (using both Postgres or Redis)
+4. Proxy Routing Configuration (using EnvoyProxy itself)
+
+it includes a Dockerfile for building the image and a docker-compose file for running the container.
+
+## Pre-requisites
+
+### Postgres
+```
+docker run --name postgres-envoy-test -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres
+```
+
+Run init script to create the database and the table
```

Comment:
Consideration for later, we can make the envoy proxy have read only permissions to the db or cache, while the token manager preserves the write permissions